frontend/src/metabase/meta/Parameter.js

```diff
@@ -95,33 +185,36 @@ function fieldFilterForParameter(parameter: Parameter) {
 function fieldFilterForParameterType(
   parameterType: ParameterType,
 ): FieldPredicate {
-  const [type] = parameterType.split("/");
+  const [type] = splitType(parameterType);
   switch (type) {
     case "date":
       return (field: Field) => field.isDate();
     case "id":
       return (field: Field) => field.isID();
     case "category":
       return (field: Field) => field.isCategory();
+    case "location":
+      return (field: Field) =>
+        field.isCity() ||
+        field.isState() ||
+        field.isZipCode() ||
+        field.isCountry();
+    case "number":
+      return (field: Field) => field.isNumber();
```

Comment:
this especially would be a good candidate for the shared lib. "What are the filter types I can use on a Field?" is something we should define there. It's useful in the backend too for stuff like generating automagic dashboards

Reply:
not saying we should change all this stuff that was already written -- especially if it's working and has tests -- but in the future we should put this stuff in the shared lib